vendor/clangir/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+#
+# Copyright (c) 2025-present, Trail of Bits, Inc.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+#
+
+set(src_dir "${CMAKE_CURRENT_LIST_DIR}/src")
+set(build_dir "${CMAKE_CURRENT_BINARY_DIR}/build")
+
+# Fetch the submodule if we don't yet have it.
+if(NOT EXISTS "${src_dir}/llvm/CMakeLists.txt")
+  execute_process(
+    COMMAND "${GIT_EXECUTABLE}" submodule update --init vendor/llvm-project/src
+    WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}")
+
+  execute_process(
+    COMMAND "${GIT_EXECUTABLE}" submodule set-branch --branch pasta-llvmorg-17.0.4 vendor/llvm-project/src
```

Comment:
(Asking without judgement here - primarily seeking to understand) what does this mean? Does this mean we're dependent on 2 different llvm version branches that aren't merged, ie of LLVM20 elsewhere and of LLVM17.0.4 here? 🤔